apps/ledger-live-mobile/src/families/evm/EditTransactionFlow/EditTransactionSummary.tsx

```diff
@@ -0,0 +1,348 @@
+import { accountScreenSelector } from "../../../reducers/accounts";
+import SummaryAmountSection from "../../../screens/SendFunds/SummaryAmountSection";
+import SummaryFromSection from "../../../screens/SendFunds/SummaryFromSection";
+import SummaryNft from "../../../screens/SendFunds/SummaryNft";
+import SummaryToSection from "../../../screens/SendFunds/SummaryToSection";
+import SummaryTotalSection from "../../../screens/SendFunds/SummaryTotalSection";
+import { CurrentNetworkFee } from "../CurrentNetworkFee";
+import { EditTransactionParamList } from "./EditTransactionParamList";
+
+type Navigation = BaseComposite<
+  StackNavigatorProps<EditTransactionParamList, ScreenName.EditTransactionSummary>
+>;
+
+type Props = Navigation;
+
+function EditTransactionSummary({ navigation, route }: Props) {
+  const { colors } = useTheme();
+  const { nextNavigation, overrideAmountLabel, transactionRaw, editType } = route.params;
+
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+
+  invariant(account, "account is missing");
+  invariant(transactionRaw, "transactionRaw is missing");
+
+  const {
+    transaction,
+    setTransaction,
+    status: txStatus,
+    bridgePending,
+  } = useBridgeTransaction(() => ({
+    transaction: route.params.transaction,
+    account,
+    parentAccount,
+  }));
+
+  const transactionToUpdate = fromTransactionRaw(transactionRaw) as EvmTransaction;
+
+  const status = getEditTransactionStatus({
+    transaction: transaction as EvmTransaction,
+    transactionToUpdate,
+    status: txStatus,
+    editType,
+  });
+
+  invariant(transaction, "transaction is missing");
+
+  const isNFTSend = isNftTransaction(transaction);
+
+  // handle any edit screen changes like fees changes
+  useTransactionChangeFromNavigation(setTransaction);
+
+  const [highFeesOpen, setHighFeesOpen] = useState(false);
+
+  const navigateToNext = useCallback(() => {
+    if (!nextNavigation) {
+      return;
+    }
+    return (
+      // This component is used in a wild bunch of navigators.
+      // nextNavigation is a param which can have too many shapes
+      // Unfortunately for this reason let's keep it untyped for now.
+      (navigation as StackNavigationProp<{ [key: string]: object }>).navigate(nextNavigation, {
+        ...route.params,
+        transaction,
+        status,
+        selectDeviceLink: true,
+      })
+    );
+  }, [navigation, nextNavigation, route.params, transaction, status]);
+
+  const onAcceptFees = useCallback(() => {
+    setHighFeesOpen(false);
+    navigateToNext();
+  }, [navigateToNext]);
+  const onRejectFees = useCallback(() => {
+    setHighFeesOpen(false);
+  }, []);
+
+  const { amount, totalSpent, errors, warnings } = status;
+
+  const firstError = errors[Object.keys(errors)[0]];
+
+  const mainAccount = getMainAccount(account, parentAccount);
+  const currencyOrToken = getAccountCurrency(account);
+
+  const hasNonEmptySubAccounts =
+    account &&
```

Comment:
Don't need this check as you have the invariant line 63 right ?

Reply:
Indeed, some checks are unnecessary here

3ef6e05d4ec4ab99dd05bdde7db4597ea234eac3